docs/manual/spec.md

```diff
@@ -471,6 +471,58 @@ In simple packages `%prep` is often just:
 %prep
 %autosetup
 ```
+
+#### %setup
+
+`%setup [options]`
+
+The primary function of `%setup` is to set up the build directory for the
+package, typically unpacking the package's sources but optionally it
+can just create the directory. It accepts a number of options:
+
+```
+-a N        unpack source N after changing to the build directory
+-b N        unpack source N before changing to the build directory
+-c          create the build directory (and change to it) before unpacking
+-D          do not delete the build directory prior to unpacking (used
+            when more than one source is to be unpacked with `-a` or `-b`)
+-n DIR      set the name of build directory (default is `%{name}-%{version}`)
+-T          skip the default unpacking of the first source (used with
+            `-a` or `-b`)
+-q          operate quietly
+```
+
+#### %patch
+
+`%patch [options] [arguments]`
+
+`%patch` is used to apply patches on top of the just unpacked pristine sources.
+Historically it supports multiple strange syntaxes which are nowadays
+deprecated. To apply patch number 1, the main syntaxes are:
```

Comment:
I’m curious what the others are :smile:.

Reply:
See https://github.com/rpm-software-management/rpm/blob/3e8c997a281031b40dd8412d745c8b0e4580c788/build/parsePrep.c#L379 for starters, but it's the combos that really kill: "%patch 1" would try to apply patch number 0 and 1. And so on.